### PR TITLE
[LoongArch64] Add some R2R missing codes.

### PR DESCRIPTION
**Pull request by LuckyXu-HF**
[LoongArch64] Add some R2R missing codes for LA64.

### CODE REVIEW

**Comment by am11** [comment]
Looks good! Please also add corresponding bits in `src/tasks/Crossgen2Tasks/ResolveReadyToRunCompilers.cs`. Once https://github.com/dotnet/installer/pull/19469 is picked up (preview 4 update in `global.json` when it's [released](https://dotnet.microsoft.com/en-us/download/dotnet/9.0)), we can enabled LA64 support in src/installer; similar to #96941.

File: src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/ManagedTextSection.cs [hunk 1]
```diff
@@ -90,7 +90,7 @@ internal sealed class ManagedTextSection
         /// If set, the module contains instructions that assume a 64 bit instruction set. For example it may depend on an address being 64 bits.
         /// This may be true even if the module contains only IL instructions because of PlatformInvoke and COM interop.
         /// </summary>
-        internal bool Requires64bits => Machine == Machine.Amd64 || Machine == Machine.IA64 || Machine == Machine.Arm64;
+        internal bool Requires64bits => Machine == Machine.Amd64 || Machine == Machine.IA64 || Machine == Machine.Arm64 || Machine == Machine.LoongArch64;
```
Comment (jkotas) [inline src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/ManagedTextSection.cs:93]: These checks in System.Reflection.Metadata are only needed for platforms supported by .NET Framework on Windows.

Could you please revert the changes in System.Reflection.Metadata? 

If you need these changes to make something work, let's discuss how to address it differently.

**Comment by LuckyXu-HF** [comment]
> Looks good! Please also add corresponding bits in `src/tasks/Crossgen2Tasks/ResolveReadyToRunCompilers.cs`. Once [dotnet/installer#19469](https://github.com/dotnet/installer/pull/19469) is picked up (preview 4 update in `global.json` when it's [released](https://dotnet.microsoft.com/en-us/download/dotnet/9.0)), we can enabled LA64 support in src/installer; similar to #96941.

Hi @am11 ，big thanks for your reminder, I have updated the corresponding bits in `src/tasks/Crossgen2Tasks/ResolveReadyToRunCompilers.cs` for LA64.

Note that the R2R-function in the main-dev branch temporarily is not as stable as our released 8.0 branch, may be some bugs still exists in main which we will continue to track.

@shushanhf  Could you please review this PR? Thanks.

File: src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/ManagedTextSection.cs (hunk 1 above)
Comment (LuckyXu-HF) [reply src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/ManagedTextSection.cs]: Ok, thank you very much, I have revert the change in `ManagedTextSection.cs`. As you said, revert here will not cause any impact on linux-LA64.
But the `PEHeaderBuilder.cs` which also under the directory `System.Reflection.Metadata` is important, this will affect the packaging format for the dll under R2R.

**Review by shushanhf — APPROVED** [review]
LGTM
Thanks

**Comment by LuckyXu-HF** [comment]
Hi @jkotas , could you please review this PR? Thanks.

**Review by jkotas — APPROVED** [review]
Thanks

**Comment by jkotas** [comment]
/ba-g Build analysis not recognizing known failure. See https://github.com/dotnet/runtime/issues/101567 for details.
